### PR TITLE
Some changes I had to do to make an webapp based on this library

### PR DESCRIPTION
**Pull request by evertheylen**
While creating [this little app](https://github.com/evertheylen/notion-export-ics), there were a few problems I had:

  - I found the library a bit too slow. I identified two issues: the extra requests generated by the `Monitor` and the extra requests needed to identify a `collection` for a given `RowBlock`. I solved this by (re)introducing the `monitor` keyword, and adding the `collection` attribute manually where it makes sense to do so. These two changes cut the amount of requests down from _O(n)_ to _O(1)_, where _n_ is the amount of calendar events. Needless to say, it's a drastic speedup :)
  - I needed the `calendar_by` attribute. The solution is obvious, but I'm not sure whether this is the best way to do this.
  - While deploying, it was kind of problematic that the library specifically wants to put its cache in the user's home directory. I solved it with an environment variable, but I'm not sure whether this is the best way. Maybe this should not be a global variable at all?

As this is a draft PR, I don't think this is ready to be merged as-is, but I'm looking for other people's opinions (in particular @jamalex of course :) ).

### CODE REVIEW

**Comment by evertheylen** [comment]
Not sure if github notified you @jamalex, but maybe it's because of the draft thing. Either way, I'm still looking forward to your feedback :)

File: notion/client.py [hunk 1]
```diff
@@ -26,14 +26,15 @@ class NotionClient(object):
     for internal use -- the main one you'll likely want to use is `get_block`.
     """
 
-    def __init__(self, token_v2, start_monitoring=True, cache_key=None):
+    def __init__(self, token_v2, monitor=True, start_monitoring=True, cache_key=None):
         self.session = Session()
         self.session.cookies = cookiejar_from_dict({"token_v2": token_v2})
         cache_key = cache_key or hashlib.sha256(token_v2.encode()).hexdigest()
         self._store = RecordStore(self, cache_key=cache_key)
-        self._monitor = Monitor(self)
```
Comment (rampeer) [inline notion/client.py]: It's a good practice to set the variables to None if they are not used. So, I would suggest to set `self._monitor` to `None` if `monitor` flag is off (right now, it won't be set at all).

This way, all code can assume that the attribute `_monitor` exists, and work with that. No clumsy checks like `if hasattr(self, '_monitor')` won't be required anywhere in the future.

**Comment by ivanistheone** [comment]
I can confirm this fixes the 400 errors. I was able to use notion API last couple of days thanks to the code fixes in this PR.

**Comment by Leechael** [comment]
Please approve because current is broken and this PR can rescue.

**Comment by ivanistheone** [comment]
@Leechael In the meantime you create a file `requirements.txt` with contents
```
git+https://github.com/evertheylen/notion-py@master
```
then install using `pip install -r requirements.txt` to get the code with the PR applied.

**Comment by Leechael** [comment]
@ivanistheone yes current I'm did it this way.

**Pull request merged**